api/tableau_api/views.py

```diff
@@ -8,9 +69,180 @@ class SummaryStatisticDailyStudyView(TableauApiView):
+                errs.append(e)
+        if errs:
+            raise ValidationError(errs, code='invalid_choice')
+
+
+class MultiErrorMultipleChoiceField(forms.MultipleChoiceField):
+    def validate(self, value):
+        """Validate that the input is a list or tuple."""
+        if self.required and not value:
+            raise ValidationError(self.error_messages['required'], code='required')
+        # Validate that each value in the value list is in self.choices.
+        errors = []
+        for val in value:
+            if not self.valid_value(val):
+                errors.append(ValidationError(
+                    self.error_messages['invalid_choice'],
+                    code='invalid_choice',
+                    params={'value': val},
+                ))
+        if errors:
+            raise ValidationError(
+                errors,
+                code='invalid_choice'
+            )
+
+
+class ApiQueryForm(forms.Form):
+    # study_id is cleaned to the object_id of the chosen Study on output
+    study_id = forms.ModelChoiceField(queryset=Study.objects.all(),
+                                      required=True,
+                                      to_field_name="object_id",
+                                      error_messages={'invalid_choice': "no study with that id was found"})
+
+    end_date = forms.DateField(required=False,
+                               error_messages={'invalid': "end date could not be interpreted as a date. Dates should be"
+                                                          "formatted as 'mm/dd/yyyy' (without quotes)"})
+
+    start_date = forms.DateField(required=False,
+                                 error_messages={
+                                     'invalid': "start date could not be interpreted as a date. Dates should be"
+                                                "formatted as 'mm/dd/yyyy' (without quotes)"})
+
+    limit = forms.IntegerField(required=False,
+                               error_messages={'invalid': "limit value could not be interpreted as an integer value"})
+
+    ordered_by = forms.ChoiceField(choices=[(f, f) for f in field_names],
+                                   required=False,
+                                   error_messages={'invalid_choice': "%(value)s is not a field that can be used "
+                                                                     "to sort the output"})
+
+    order_direction = forms.ChoiceField(choices=[('ascending', 'ascending'), ('descending', 'descending')],
+                                        required=False,
+                                        error_messages={'invalid_choice': "If provided, the order_direction parameter "
+                                                                          "should contain either the value 'ascending' "
+                                                                          "or 'descending'"})
+
+    participant_ids = CsvField(required=False)
```

Comment:
We should rename this to `participant_patient_id` so it's less confusing from a developer perspective, and just add a `name='participant_id` to the params.